src/main/java/com/bobocode/svydovets/beans/scanner/ConfigurationBeanScanner.java

```diff
@@ -59,7 +61,19 @@ public Map<String, BeanDefinition> scan(String packageName) {
 
     @Override
     public void fillDependsOn(Map<String, BeanDefinition> beanDefinitions) {
-        //TODO implement
+        beanDefinitions.values().stream()
+                .filter(bd -> Objects.nonNull(bd.getFactoryMethod()))
+                .forEach(bd -> fillDependsOn(beanDefinitions, bd));
+    }
+
+    private void fillDependsOn(Map<String, BeanDefinition> beanDefinitions, BeanDefinition bd) {
+        var dependsOn = new ArrayList<>();
+
+        var parameterTypes = bd.getFactoryMethod().getParameterTypes();
+        Arrays.stream(parameterTypes)
+                .forEach(type -> dependsOn.add(findDependsOnByClass(beanDefinitions, bd.getBeanClass(), type)));
+
+        bd.setDependsOn(dependsOn.toArray(new String[0]));
```

Comment:
Do we set an empty array?

Reply:
Yes, toArray will create array of appropriate size if provided array is smaller than the list. This change Idea is suggested.